### DISCUSSION TITLE
[ Common ] Incorrect Button Mapping despite being correct in launcher

### DISCUSSION THREAD

**Discussion by icegoddessLexra** [julianxhokaxhiu/FFNx#308] [Q&A] [answered]
Today I went through the installation process with FFNx for the 2013 Steam EN version of FFVIII to replace the midi files with the original PS tracks. I followed the install instructions in the guide posted in the github, the version I downloaded was the FFNx-Steam-v1.10.0.25 version. I initially tried the v1.10.0.0 version but my game wouldn't even load the launcher, it'd crash. So, I restored the game files and tried with the pre-release branch. I was able to get the custom music working but at a rather severe cost: my controller inputs got messed up. 

I'm able to map my controller in the launcher, but then those inputs don't seem to carry over correctly when I load the game up. I set my controls (in the FF8 launcher) to be vanilla to what the PS1 version controls were, and I'm using a wired PS4 controller. I had no issues with my controller inputs being correct _before_ installing the music mod.

But now, even though I have Select/Confirm set to 'X', Cancel to 'O' they're incorrect when I boot the game. X is now cancel, Square is Confirm, and O just makes me walk. Pause is now my R2, despite being set to my controller's pause button.

In my FFNx.toml file under the controllers section it was originally set to: steam_game_userdata = "". I've since added in my userfile so it reads: steam_game_userdata = 'user_4075028'. And there was no change to how the game views my control schematics. Is there a setting in the file I'm missing that's messing with my button schematics? Or is this a bug?

I'm suddenly stuck with confirm being Square, Cancel being X, and all the other incorrect inputs. It seems the only buttons mapped correctly still are Triangle for the Menu and R1 being Squall's gunblade trigger. Analog and the D-pad inputs also work correctly.
When I removed the music mod my controller inputs return to what they're supposed to be

**Comment by julianxhokaxhiu** [comment, marked as answer]
Control mapping on Steam is not supported. You need to reconfigure the controls in your game, and by default the controls in relatively:
-> FF7: `ff7input.cfg`
-> FF8: `ff8input.cfg`

Will be used. Anything else on top is totally up to your discretion.
